build/accelerate_launch.py

```diff
@@ -79,8 +84,27 @@ def main():
                 if actions_type_map.get(key) == "_StoreAction":
                     accelerate_launch_args.append(str(val))
 
-    num_processes = accelerate_config.get("num_processes")
+    # accept setting SET_NUM_PROCESSES_TO_NUM_GPUS=True in Shell interpreted as string
+    set_num_processes_to_num_gpus = os.getenv(
+        "SET_NUM_PROCESSES_TO_NUM_GPUS", "True"
+    ).lower()
+    user_arg_num_processes = accelerate_config.get("num_processes")
+    num_processes = 0
+    if set_num_processes_to_num_gpus == "true":
+        num_processes = torch.cuda.device_count()
```

Comment:
should be number of GPUs available in node? what we specify for that Pytorch training job resource should be the number of GPUs also available.
Is there a concern here? 
_(tagging @tharapalanivel as well )_

Reply:
It is the number of GPUs available in the deployment configuration. I tested on devstage cluster and on research cluster setting:
```yaml
      resources:
        limits:
          nvidia.com/gpu: "1"
```
and `torch.cuda.device_count()` gives 1. Same when I increase this value to 2,3,4.

Reply:
> should be number of GPUs available in node? 

It should not be number of nodes available in the node, since we may not be wanting to use all the GPUs, and the deployment config might be defining smaller number. 

> It is the number of GPUs available in the deployment configuration. 

Perfect! Thanks for validating!

Reply:
Thanks Gaurav and Anh! Sorry I was confusing node and pod - understood now . In the resources limit we specify limit of GPUs for each replica, and within that pod when we run nvidia commands we only see resources we specified . Got it